### ISSUE TITLE
Accordion

### ISSUE THREAD

**Issue by 000-250-006** [Free-Cities/Free-Cities#134]
I'm working a revision of accordion to fix one bug (in the game startup options) and some cosmetic expansions (Arcology reports for one)

Are there any other displays that would benefit from being able to reduce their screen real estate footprint with accordion?

**Comment by freecitiesdev** [comment]
To be completely honest, I haven't had the time to closely review exactly
what your code *does*. The past week has been extremely tight for time, so
I mostly concentrated on making sure it *worked*. Would you give me the
short summary?

FCD

**Comment by 000-250-006** [comment]
It uses CSS to collapse/reveal long blocks of text. I came up with it for end-of-week reports originally, because I wanted to be able to see details but not have to wade through very long blocks of text to find the category I was looking for.

Purely cosmetic feature, part of a personal project to make FC relatively 508 compliant ...

I make pretty things beautiful :)

**Comment by freecitiesdev** [comment]
>
> 508 compliant


Explain further. The only 508 compliance I know of is for folks with
disabilities.

FCD

**Comment by 000-250-006** [comment]
Precisely.

You went and created this amazing text based story-driven interactive game. People who use assistive tech get horny too. Accessible UX is kind of my specialty, so ...

**Comment by freecitiesdev** [comment]
Maybe I'm missing something obvious, but what's the connection between
collapsing text and accessibility? Don't get me wrong, I'm 110% in favor of
as much accessibility as possible, in FC and elsewhere.

**Comment by 000-250-006** [comment]
Short answer: density.

**Comment by Folant** [comment]
@000-250-006 Just in case:
[This line](https://github.com/Free-Cities/Free-Cities/blob/master/src/js/accordionJS.tw#L19) should be ```Array.prototype.slice.call(document.querySelectorAll(".macro-include"))``` now because of #131

**Comment by 000-250-006** [comment]
Thanks! I knew something weird had changed. Patch to come

**Comment by 000-250-006** [comment]
Hopefully fix in #143 tyvm

**Comment by 000-250-006** [comment]
OK, got some feedback on the latest iteration, I think it's now at the stage where everything it can be useful for has been identified.

So latest ver pull request closed, I should have the final version finished this weekend. It will be a radical rewrite of the code from dev long-form stepping to condensed, so don't be all freaked out at the extent of the changes :)

**Comment by Bane70** [comment]
Are they going to include any of the facility reports do you know? As I'm working on trying to reformat the texts a bit better and some minor bug fixes.

**Comment by 000-250-006** [comment]
Yeah, they've got text formatting tweaks in them too ...

**Comment by 000-250-006** [comment]
@Bane70 see #161 ... ICYMI it looks like _[var] is temporary for the page load, not the function/conditional it relates to. Might be worth adding <<unset _[whatever]>> when you're done with a loop to clear it out

**Comment by Bane70** [comment]
Yeah the _temps stay active till a goto, links/clicks, and maybe something else clears them as well, before they get wiped from memory. So they are actually more like semi global _vars as I could create a _temp in Endweek and if it was not using a goto or link when the code came back to slaveAssignmentsReport that _var would be usable there also from that point on. 

Yeah I could clear them at the end of the passage/loop etc. or leave them, not sure what would be best. Using unset's might cause the game to do more garbage collections causing the game to pause more often.

When I was digging around in the Sugarcube code it takes everything and converts it to JavaScript, so that means that the multiple conditions in an if statement use short circuiting (&&. ||). it also means things like is, isnot, eq, and, or, etc. also take a bit of time changing over to JavaScripts !==, ==, &&, ||, etc.

**Comment by 000-250-006** [comment]
That last had me banging my head on my desk a few times doing the loop for end of week. I keep defaulting back to real syntax, but Sugarcube seems to throw a fit with math that way sometimes.. Perks of a translated macro language I guess.

Gonna have to rethink a few ideas with this new info.

**Comment by Bane70** [comment]
Too bad Sugarcube doesn't take the whole file and convert it over right away. Instead it does it line by line, except for things like, << switch >><< /switch>> everything in the block between the pair gets converted over as far as I can tell. same with anything within the << .... >>. That's why if there are multiple << set >>'s in a row I change them over to a single one. as it cuts down on Sugarcube calling/finding set's function more than once and having to convert over to JS multiple times.

**Comment by 000-250-006** [comment]
Be happy, I hear native Twine doesn't even guarantee execution order :P

I'm constantly amazed at @freecitiesdev's perseverance getting this working as they have, TBH. The more I delve into Sugarcube working out how to jack the accessibility factor up, the more I wonder why they didn't just throw up their hands a long time ago :P

Doesn't help that "my" language is PHP. I'm not used to these ███████ pieces of ████ "interpreted" languages, let alone macros. It's like going from *IX to Windows 10!

I want my classes, and I want them now, dammit!!!!!!

</rant> :)

**Comment by fcpmanon** [comment]
#456 added two additional reports to the accordion:  Penthouse Report and Rules Assistant Report.

I would have liked to "redirect" some text for inclusion in later reports, specifically the text generated by "SA choose own job", currently printed in the Penthouse Report, would ideally show up at the top of the facility report for the facility where they decided to go.

Is there a limit to the amount of text that can be collapsed into an accordion?
